src/next-collection-view.js

```diff
@@ -0,0 +1,615 @@
+      });
+    }
+
+    return childView;
+  },
+
+  // First check if the `view` is a view class (the common case)
+  // Then check if it's a function (which we assume that returns a view class)
+  _getView(view, child) {
+    if (view.prototype instanceof Backbone.View || view === Backbone.View) {
+      return view;
+    } else if (_.isFunction(view)) {
+      return view.call(this, child);
+    }
+  },
+
+  _getChildViewOptions(child) {
+    if (_.isFunction(this.childViewOptions)) {
+      return this.childViewOptions(child);
+    }
+
+    return this.childViewOptions;
+  },
+
+  // Build a `childView` for a model in the collection.
+  // Override to customize the build
+  buildChildView(child, ChildViewClass, childViewOptions) {
+    const options = _.extend({model: child}, childViewOptions);
+    return new ChildViewClass(options);
+  },
+
+  _setupChildView(view) {
+    monitorViewEvents(view);
+
+    // set up the child view event forwarding
+    this.listenTo(view, 'all', this._childViewEventHandler);
+
+    // We need to listen for if a view is destroyed in a way other
+    // than through the CollectionView.
+    // If this happens we need to remove the reference to the view
+    // since once a view has been destroyed we can not reuse it.
+    view.on('destroy', this.removeChildView, this);
+
+    view._parent = this;
+  },
+
+  // used by `_childViewEventHandler`
+  _getImmediateChildren() {
+    return this.children._views;
+  },
+
+  // Overriding Backbone.View's `setElement` to handle
+  // if an el was previously defined. If so, the view might be
+  // attached on setElement.
+  setElement() {
+    const hasEl = !!this.el;
+
+    Backbone.View.prototype.setElement.apply(this, arguments);
+
+    if (hasEl) {
+      this._isAttached = isNodeAttached(this.el);
+    }
+
+    return this;
+  },
+
+  // Render children views.
+  render() {
+    this._ensureViewIsIntact();
+    this.triggerMethod('before:render', this);
+
+    this._destroyChildren();
+
+    // After all children have been destroyed re-init the container
+    this._initChildViewStorage();
+
+    this._addChildModels(this.collection.models);
+
+    this._showChildren();
+
+    this._isRendered = true;
+
+    this.triggerMethod('render', this);
+    return this;
+  },
+
+  sort() {
+    this._showChildren();
+    return this;
+  },
+
+  _showChildren(removedViews = []) {
+    if (this.isEmpty()) {
+      this._showEmptyView();
+      return;
+    }
+
+    this._destroyEmptyView();
+
+    this.triggerMethod('before:sort', this);
+
+    this._sortChildren();
+
+    const filteredViews = this._filterChildren(removedViews);
+
+    this._detachChildren(filteredViews.detaching);
+
+    // Check if all views have been filtered out
+    if (this.isEmpty(!filteredViews.attaching.length)) {
+      this._showEmptyView()
+    } else {
+      this._renderChildren(filteredViews.attaching);
+    }
+
```

Comment:
@scott-w This might be a good place to throw a `update:children' that get's passed `filteredViews` so you know what views were attached/detached.  Again though.. it's possible the no views were added or removed if this was just a sort.. or a filter was applied that made no changes.

Unlike `remove:child` the views will not have been destroyed yet, and some of them may just be detached because of a filter.